File: SignalRDashboard.Data.Milliman/Hubs/Models/TwitterStatus.cs
using SignalRDashboard.Data.Core.Hubs.Models;

namespace SignalRDashboard.Data.Milliman.Hubs.Models
{
    public class TwitterStatus : DashboardHubModel
    {
        private string _lastTweet;
        public string LastTweet
        {
            get { return _lastTweet; }
            set { SetProperty(ref _lastTweet, value); }
        }
        private bool _tweetIsNew;
        public bool TweetIsNew
        {
            get { return _tweetIsNew; }
            set { SetProperty(ref _tweetIsNew, value); }
        }
        private string _lastTweetTime;
        public string LastTweetTime
        {
            get { return _lastTweetTime; }
            set { SetProperty(ref _lastTweetTime, value); }
        }
    }
}

File: SignalRDashboard.Web/Scripts/app/models/milliman/TwitterStatus.js
(function(signalrdashboardMilliman) {
    signalrdashboardMilliman.TwitterStatus = TwitterStatus;

    function TwitterStatus() {
        // Properties matching data model received via SignalR
        this.lastTweet = '';
        this.lastTweetTime = '';
        this.tweetIsNew = false;

        // Error tracking properties
        this.hasNewTweet = false;

        // Error check conditions
        var model = this;
        var errorChecks = [];
        errorChecks.push({ model: model, condition: function() { return model.tweetIsNew; }, targetProperty: 'hasNewTweet' });

        this.updateFromData = function(data) {            
            signalrdashboard.mapping.map(data, this);
            signalrdashboard.alertHandler.performErrorChecks('TwitterStatus', errorChecks);
        };
    }
    
})(window.signalrdashboard.milliman || (window.signalrdashboard.milliman = {}));